### PR TITLE
Fix snowflake connector SQL util (to_pandas issue)  

### PR DESCRIPTION
**Pull request by sfc-gh-dhuang**
# Description

Please include a summary of the changes and the related issue that can be
included in the release announcement. Please also include relevant motivation
and context.

## Other details good to know for developers

Please include any other details of this change useful for _TruLens_ developers.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] New Tests
- [ ] This change includes re-generated golden test results
- [ ] This change requires a documentation update

----

> [!IMPORTANT]
> Fix `execute_query` in `sql_utils.py` to correctly handle non-SELECT queries and add comprehensive unit tests in `test_snowflake_sql_utils.py`.
> 
>   - **Bug Fix**:
>     - Fix `execute_query` in `sql_utils.py` to use `to_pandas` only for SELECT queries, and `collect` for others.
>   - **Tests**:
>     - Add `test_snowflake_sql_utils.py` with unit tests for `execute_query` covering SELECT, non-SELECT, parameterized queries, and error handling.
>     - Tests include column name cleanup and case-insensitive query handling.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=truera%2Ftrulens&utm_source=github&utm_medium=referral)<sup> for 15c9661ad1ddb14a7cd934043883da9e8f4591c8. You can [customize](https://app.ellipsis.dev/truera/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: src/connectors/snowflake/trulens/connectors/snowflake/dao/sql_utils.py [hunk 1]
```diff
@@ -52,18 +52,17 @@ def execute_query(
     Executes a query with optional parameters with qmark parameter binding (if applicable).
     """
     try:
-        # For SHOW commands, Snowpark has a bug where plan attributes don't match actual results
-        # Use collect() and manually construct DataFrame to avoid schema mismatch
-        if query.strip().upper().startswith("SHOW"):
+        if query.strip().upper().startswith("SELECT"):
+            # snowpark to_pandas supports only SELECT statements up until recent releases https://github.com/snowflakedb/snowpark-python/blob/main/CHANGELOG.md
+            df = session.sql(query, params=parameters).to_pandas()
```
Comment (sfc-gh-dhuang) [inline src/connectors/snowflake/trulens/connectors/snowflake/dao/sql_utils.py:57]: note: not every SELECT query in snow sql has to start with SELECT, i.e. A SELECT query using CTE can start with `WITH`. 

However, the `.collect` fallback below is still going to work in these edge cases.

File: tests/unit/test_snowflake_sql_utils.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+import pandas as pd
+import pytest
+
+try:
+    from snowflake.snowpark.row import Row
+    from trulens.connectors.snowflake.dao.sql_utils import execute_query
+
+except ImportError:
+    pytest.skip("execute_query is not available", allow_module_level=True)
```
Comment (sfc-gh-dkurokawa) [inline tests/unit/test_snowflake_sql_utils.py:12]: why would this not be available?
